Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +38,136 @@ namespace
         }
     }
 
+    bool EndsWith(std::string_view s, char c)
+    {
+        if (!s.empty())
+        {
+            return s.back() == c;
+        }
+        return false;
+    }
+
+    // based on: https://stackoverflow.com/a/26221725
+    template<typename T>
+    std::string FormatValue(std::string& formatString, T value)
+    {
+        // get the size of what will be written (+1 for null terminator)
+        int size_s = snprintf(nullptr, 0, formatString.c_str(), value) + 1;
+        // if size is 0 or less, we had a formatting error
+        if (size_s <= 0)
+        {
+            throw std::runtime_error("Error during console.log formatting.");
+        }
+        size_t size = static_cast<size_t>(size_s);
+        // create a string to hold that size (-1 as we don't want the null terminator)
+        std::string buf(size - 1, '0');
```

Comment:
nit: Modern C++ uses brace initialization. Many places.
```suggestion
        std::string buf{size - 1, '0'};
```